File: src/cpp/dynamic_array_with_allocated.cpp
#include <vector>
#include <iostream>
#include <string>

int main(int argc, char *argv[]){
    // Intend to be called like ./dynamic_array_with_builtin 5
    int pushedItems = std::atoi(argv[argc-1]);
    std::cout << "Testing a push/pop with " << pushedItems << " allocated items.\n";

    std::vector<std::string> myvec {};
    for (int i = 0; i < pushedItems; i++)
    {
        myvec.push_back(std::string{std::to_string(i)});
    }
    for (auto i : myvec)
    {
        myvec.pop_back();
    }
}

File: src/lua/dynamic_array_with_allocated.lua
itemCount = tonumber(arg[1],10)
print(string.format("Testing a push/pop of %d allocated items.", itemCount))
charcount = 0
table = {}
for arrayVal = 0,itemCount do
    table[arrayVal] = tostring(arrayVal)
end

for arrayVal = 0,itemCount do
    charcount = charcount + string.len(table[arrayVal])
    table[arrayVal] = nil
end

print(string.format("There were %d characters written", charcount))